src/test/java/org/cactoos/scalar/XorTest.java

```diff
@@ -0,0 +1,169 @@
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.ScalarHasValue;
+
+/**
+ * Test case for {@link Xor}.
+ * @since 0.48
+ */
+@SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.TooManyMethods"})
+final class XorTest {
+
+    @Test
+    void trueTrue() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new True(),
+                new True()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void falseTrue() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new False(),
+                new True()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void treuFalse() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new True(),
+                new False()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void falseFalse() throws Exception {
+        new Assertion<>(
+            "Either one, but not both nor none",
+            new Xor(
+                new False(),
+                new False()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void singleTrue() throws Exception {
+        new Assertion<>(
+            "Single True must be True",
+            new Xor(
+                new True()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void singleFalse() throws Exception {
+        new Assertion<>(
+            "Single False must be False",
+            new Xor(
+                new False()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void emptyIterator() throws Exception {
+        new Assertion<>(
+            "Empty iterator must be true",
+            new Xor(new IterableOf<Scalar<Boolean>>()),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void oddNumberOfTrue() throws Exception {
+        new Assertion<>(
+            "Odd number of True must be True",
+            new Xor(
+                new False(),
+                new False(),
+                new True()
+            ),
+            new ScalarHasValue<>(true)
+        ).affirm();
+    }
+
+    @Test
+    void evenNumberOfTrue() throws Exception {
+        new Assertion<>(
+            "Even number of True must be False",
+            new Xor(
+                new False(),
+                new True(),
+                new True()
+            ),
+            new ScalarHasValue<>(false)
+        ).affirm();
+    }
+
+    @Test
+    void allFalse() throws Exception {
+        new Assertion<>(
+            "Even number of True must be False",
```

Comment:
@kokodyn the message is incorrect here I think

Reply:
@victornoel fixed

Reply:
@kokodyn you apparently changed the wrong test, the one you changed (`oddNumberOfTrue`) was correct. Can you fix both of them now?

Reply:
@victornoel  thanks for comment, problem fixed.